### PR TITLE
gajim: 1.0.3 -> 1.1.2

### PR DESCRIPTION
**Pull request by aszlig**
The list of upstream changes is huge, so I'm not pasting it here in the commit message, but here is the upstream URL:

  https://dev.gajim.org/gajim/gajim/blob/gajim-1.1.2/ChangeLog

One of the most visible updates are the design changes for various dialogs and the Emoji overhauls.

On our end, we now need three more dependencies, namely `cssutils`, `precis-i18n` and `keyring`, which I added accordingly.

In addition, the test runner is now integrated into setup.py, which we now use.

I also cleaned up the package expression a bit, eg. it's no longer wrapped in a big `with lib;`, so that `nix-instantiate --parse` is able to detect attribute errors (which is very useful if you have editor integration).

### CODE REVIEW

**Comment by etu** [comment]
@GrahamcOfBorg build python37Packages.precis-i18n python37Packages.nbxmpp python27Packages.nbxmpp gajim python27Packages.precis-i18n

File: pkgs/applications/networking/instant-messengers/gajim/default.nix [hunk 1]
```diff
@@ -12,16 +21,14 @@
 , extraPythonPackages ? pkgs: [], pythonPackages
 }:
 
-with lib;
-
 buildPythonApplication rec {
   name = "gajim-${version}";
```
Comment (dotlambda) [inline pkgs/applications/networking/instant-messengers/gajim/default.nix]:
```suggestion
  pname = "gajim";
```

File: pkgs/applications/networking/instant-messengers/gajim/default.nix [hunk 2]
```diff
@@ -1,7 +1,16 @@
 { buildPythonApplication, lib, fetchurl, gettext, wrapGAppsHook
+
+# Native dependencies
 , python, gtk3, gobject-introspection
-, nbxmpp, pyasn1, pygobject3, gnome3, dbus-python, pillow
+
+# Python dependencies
+, nbxmpp, pyasn1, pygobject3, gnome3, dbus-python, pillow, cssutils
+, precis-i18n, keyring
```
Comment (dotlambda) [inline pkgs/applications/networking/instant-messengers/gajim/default.nix]: Would you mind getting rid of `python3Packages.callPackage` in `all-packages.nix` and specifying `python3` as argument instead?

File: pkgs/applications/networking/instant-messengers/gajim/default.nix [hunk 3]
```diff
@@ -12,16 +21,14 @@
 , extraPythonPackages ? pkgs: [], pythonPackages
```
Comment (dotlambda) [inline pkgs/applications/networking/instant-messengers/gajim/default.nix]:
```suggestion
, extraPythonPackages ? pkgs: []
```

File: pkgs/applications/networking/instant-messengers/gajim/default.nix [hunk 4]
```diff
@@ -1,7 +1,16 @@
 { buildPythonApplication, lib, fetchurl, gettext, wrapGAppsHook
+
+# Native dependencies
 , python, gtk3, gobject-introspection
```
Comment (dotlambda) [inline pkgs/applications/networking/instant-messengers/gajim/default.nix]:
```suggestion
, python3, gtk3, gobject-introspection
```

File: pkgs/applications/networking/instant-messengers/gajim/default.nix [hunk 5]
```diff
@@ -31,37 +38,37 @@ buildPythonApplication rec {
...
+  ] ++ lib.optionals enableJingle [ farstream gstreamer gst-plugins-base gst-libav gst-plugins-ugly ]
+    ++ lib.optional enableSecrets libsecret
+    ++ lib.optional enableSpelling gspell
+    ++ lib.optional enableUPnP gupnp-igd;
 
   nativeBuildInputs = [
     gettext wrapGAppsHook
   ];
 
   propagatedBuildInputs = [
```
Comment (dotlambda) [inline pkgs/applications/networking/instant-messengers/gajim/default.nix]:
```suggestion
  propagatedBuildInputs = with python3.pkgs; [
```

File: pkgs/applications/networking/instant-messengers/gajim/default.nix [hunk 6]
```diff
@@ -31,37 +38,37 @@ buildPythonApplication rec {
...
-    nbxmpp pyasn1 pygobject3 dbus-python pillow
-  ] ++ optionals enableE2E [ pycrypto python-gnupg ]
-    ++ optional enableRST docutils
-    ++ optionals enableOmemoPluginDependencies [ python-axolotl qrcode ]
+    nbxmpp pyasn1 pygobject3 dbus-python pillow cssutils precis-i18n keyring
+  ] ++ lib.optionals enableE2E [ pycrypto python-gnupg ]
+    ++ lib.optional enableRST docutils
+    ++ lib.optionals enableOmemoPluginDependencies [ python-axolotl qrcode ]
     ++ extraPythonPackages pythonPackages;
```
Comment (dotlambda) [inline pkgs/applications/networking/instant-messengers/gajim/default.nix]:
```suggestion
    ++ extraPythonPackages python3.pkgs;
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by dotlambda** [comment]
A `disabled = !isPy3k` is missing.

**Comment by dotlambda** [comment]
I meant for precis-i18n: https://github.com/byllyfish/precis_i18n/blob/v1.0/setup.py#L55

**Comment by dotlambda** [comment]
Also, you should add the `pythonPackages.` prefix to two of the commit messages.

**Pull request merged**